heartpoints.sh

```diff
@@ -500,89 +505,166 @@ heartpoints_testUntilSuccess() { local timeoutSeconds=$1; local interval=$2; loc
     done
 }
 
-heartpoints_minikubeBuildDeployTest_help() { echo "minikubeBuild, then minikubeDeployTest"; }
-heartpoints_minikubeBuildDeployTest() {
+hp_minikubeBuildDeployTest_help() { echo "minikubeBuild, then minikubeDeployTest"; }
+hp_minikubeBuildDeployTest() {
     local shaToBuild="$(git_currentSha)"
-    local taggedImageName="$(heartpoints_minikubeTaggedImageName ${shaToBuild})"
-    heartpoints_minikubeBuild "${taggedImageName}" "${shaToBuild}"
-    heartpoints_minikubeDeployTest "${taggedImageName}"
+    local taggedImageName="$(hp_minikubeTaggedImageName ${shaToBuild})"
+    hp_minikubeBuild "${taggedImageName}" "${shaToBuild}"
+    hp_minikubeDeployTest "${taggedImageName}"
 }
 
-heartpoints_minikubeTaggedImageName() { local shaToBuild=$1
+hp_minikubeTaggedImageName() { local shaToBuild=$1
     requiredParameter "shaToBuild" "${shaToBuild}"
     local imageRepository="minikube"
-    echo "$(heartpoints_taggedImageName ${imageRepository} ${shaToBuild})"
+    echo "$(hp_taggedImageName ${imageRepository} ${shaToBuild})"
 }
 
-heartpoints_minikubeBuild_help() { echo "<taggedImageName> using minikube's docker daemon, build image and tag with minikube metadata"; }
-heartpoints_minikubeBuild() { local taggedImageName=$1; local shaToReportInHttpHeaders=$2
+hp_minikubeBuild_help() { echo "<taggedImageName> using minikube's docker daemon, build image and tag with minikube metadata"; }
+hp_minikubeBuild() { local taggedImageName=$1; local shaToReportInHttpHeaders=$2
     requiredParameter "taggedImageName" "${taggedImageName}"
     requiredParameter "shaToReportInHttpHeaders" "${shaToReportInHttpHeaders}"
-    heartpoints_pointToAndRunMinikubeDockerDaemon
-    heartpoints_pointToAndRunMinikubeDockerDaemon
-    heartpoints_buildAndTagImage "${taggedImageName}" "${shaToReportInHttpHeaders}"
+    hp_pointToAndRunMinikubeDockerDaemon
+    hp_pointToAndRunMinikubeDockerDaemon
+    hp_buildAndTagImage "${taggedImageName}" "${shaToReportInHttpHeaders}"
+}
+
+hp_minikubeDestroyEnvironment_help() { echo "if minikube dev environment is running, destroys it"; }
+hp_minikubeDestroyEnvironment() {
+    hp_minikube delete
```

Comment:
i'd personally rather have a function that calls the locally-installed minikube cli and require the user to have some knowledge of how to use the cli than to have wrapper functions such as this

Reply:
i think there's two points you could be making

- rather than have a source controlled way to call the minikube cli (or any other cli we depend on), count on a separately installed minikube cli already existing
- specifically for `hp minikubeDestroyEnvironment` you prefer that user would just use the choice minikube cli to discover and subsequently run the `delete` command

can you confirm / clarify around these :)

Reply:
the first point is not one that I'm making (though i'd be fine with that as well). it's just the second point